TELCODCS#835: RN - Ironic API served by proxy server in MB installations with a provisioning network (#50241)
openshift/openshift-docs: release_notes/ocp-4-12-release-notes.adoc

For more information, see xref:../installing/installing-troubleshooting.adoc#installation-bootstrap-gather_installing-troubleshooting[Troubleshooting installation issues].

> [id="ocp-4-12-consistent-adddress-for-ironicapi"]

Reviewer: Spelling: adddress

Reviewer 2: +1

> ==== Consistent IP address for Ironic API in bare-metal installations without a provisoning network  

Reviewer 2: Can you double check whether it should be `Ironic` or `ironic`? It seems to be a new term in our docs. But I see in Bare metal RNs refer to it as `ironic` API.

Author: It is a new term in the docs. It's mostly transparent to users, that's why it occasionally surfaces in RNs. OpenStack and other references to this open-source API all use Ironic, for example https://docs.openstack.org/api-ref/baremetal/, so I think Ironic is correct.

Reviewer 2: ah ok. I got to see a smaller case here https://docs.openstack.org/releasenotes/ironic/yoga.html#relnotes-20-1-0-stable-yoga-bug-fixes hence wasn't sure. Thanks for clarifying!

> With this update, in bare-metal installations without a provisioning network, the Ironic API service is accessible through a proxy server. This proxy server provides a consistent IP address for the Ironic API service. If the Metal3 pod that contains `metal3-ironic` relocates to another pod, the consistent proxy address ensures constant communication with the Ironic API service.

Reviewer 2: I am unsure about `Metal3` vs `metal3` too. 
I see both usages here https://docs.openshift.com/container-platform/4.11/installing/installing_bare_metal_ipi/ipi-install-post-installation-configuration.html#enabling-a-provisioning-network-after-installation_ipi-install-post-installation-configuration

Author: I think Metal3 is correct. The Kubernetes docs all reference Metal3 (and Ironic): https://metal3io.netlify.app/introduction.html
It is actually Metal³. But I am unsure if we want to replicate that superscript 3 in our docs. If it appears as Metal3 or metal3 in other places in our docs, then I think Metal3 is the correct one of these options.

Reviewer 2: Sounds great! Thanks for confirming.

> With this update, in bare-metal installations without a provisioning network, the Ironic API service is accessible through a proxy server. This proxy server provides a consistent IP address for the Ironic API service. If the Metal3 pod that contains `metal3-ironic` relocates to another pod, the consistent proxy address ensures constant communication with the Ironic API service.

Reviewer 2: ```suggestion
With this update, in bare-metal installations without a provisioning network, you can access the Ironic API service through a proxy server. This proxy server provides a consistent IP address for the Ironic API service. If the Metal3 pod that contains `metal3-ironic` relocates to another pod, the consistent proxy address ensures constant communication with the Ironic API service.
```

Author: Good point. I think the user can send API requests, however the Bare Metal Operator (BMO) is more likely to be using this API service to manage bare metal in the cluster AFAIK. There may be other components accessing the API also, so I would like to keep it a little vague, or at least focus on the change of API endpoint.
